sgkit/distance/api.py

```diff
@@ -77,27 +77,53 @@ def pairwise_distance(
 
     >>> x = np.array([[6, 4, 1,], [4, 5, 2], [9, 7, 3]])
     >>> pairwise_distance(x, metric='correlation')
-    array([[1.11022302e-16, 2.62956526e-01, 2.82353505e-03],
-           [2.62956526e-01, 0.00000000e+00, 2.14285714e-01],
-           [2.82353505e-03, 2.14285714e-01, 0.00000000e+00]])
+    array([[-4.44089210e-16,  2.62956526e-01,  2.82353505e-03],
+           [ 2.62956526e-01,  0.00000000e+00,  2.14285714e-01],
+           [ 2.82353505e-03,  2.14285714e-01,  0.00000000e+00]])
     """
-
     try:
-        metric_ufunc = getattr(metrics, metric)
+        metric_map_func = getattr(metrics, f"{metric}_map")
```

Comment:
nit, since `metric` only has 2 valid values, maybe it should be a Literal type instead of a str for better user API?

Reply:
Yes, makes sense.

Reply:
@aktech feel free to do it in a follow up PR if you like.

Reply:
I am happy to do this in this PR as well. I have been a bit swamped recently, hoping to wrap this up within couple of days.

Reply:
It seems Literal Types were introduced in Python: 3.8 and we [support 3.7](https://github.com/pystatgen/sgkit/blob/c210fae783e927a45d5b852105cf42d9b47a32e4/.github/workflows/build.yml#L16) as well. So, I don't see a way to add them in here, that's why the precommit tests have been failing.

Reply:
We already depend on `typing-extensions` so you should be able to use that for Literal types.

Reply:
Oh, great! I'll add that then in a bit.